src/hooks/useInitialValues.test.ts

```diff
@@ -333,4 +334,239 @@ describe('useInitialValues', () => {
+    } = hook.result;
+    expect(isBindingComplete).toBe(true);
+    expect(initialValues['latest_mother_hiv_status']).toBe('664AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA');
+  });
+
+  it('should fall back to encounter value if the calculated expression result is null or undefined', async () => {
+    let hook = null;
+    let formFields: Array<FormField> = [
+      {
+        label: 'Height (cm)',
+        type: 'obs',
+        required: false,
+        id: 'height',
+        questionOptions: {
+          rendering: 'number',
+          concept: '5090AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
+          answers: [],
+        },
+      },
+      {
+        label: 'Weight (Kgs)',
+        type: 'obs',
+        required: false,
+        id: 'weight',
+        questionOptions: {
+          rendering: 'number',
+          concept: '5089AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
+          answers: [],
+        },
+        validators: [],
+      },
+      {
+        label: 'BMI:Kg/M2 (Function calcBMI | useFieldValue)',
+        type: 'obs',
+        required: false,
+        id: 'bmi',
+        questionOptions: {
+          rendering: 'number',
+          defaultValue: 0,
+          concept: '1342AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
+          isTransient: true,
+          disallowDecimals: false,
+          calculate: {
+            calculateExpression: 'calcBMI(heigh,weight)',
+          },
+        },
+        validators: [],
+        questionInfo: 'this calculates BMI using calcBMI function and useFieldValue of weight and height',
+      },
+    ];
+
+    await act(async () => {
+      hook = renderHook(() =>
+        useInitialValues(
+          [...formFields],
+          encounter,
+          false,
+          {
+            encounter: encounter,
+            patient: testPatient,
+            location,
+            sessionMode: 'enter',
+            encounterDate: encounterDate,
+            setEncounterDate: jest.fn,
+            encounterProvider: '2c95f6f5-788e-4e73-9079-5626911231fa',
+            setEncounterProvider: jest.fn,
+            setEncounterLocation: jest.fn,
+            encounterRole: '',
+            setEncounterRole: jest.fn,
+          },
+          formFieldHandlers,
+        ),
+      );
+    });
+    const {
+      current: { initialValues, isBindingComplete },
+    } = hook.result;
+    expect(isBindingComplete).toBe(true);
+
+    expect(initialValues['height']).toBe(176);
+    expect(initialValues['weight']).toBe(56);
+
+    const bmiField = testEncounter.obs.find((field) => {
+      const formFieldNamespace = field.formFieldNamespace;
+      const formFieldPath = field.formFieldPath;
+            if (formFieldNamespace && formFieldPath.startsWith(formFieldNamespace)) {
+        const valueAfterNamespace = formFieldPath.slice(formFieldNamespace.length + 1);
+        return valueAfterNamespace === "bmi";
+      }
+      
+      return false;
+    });
+    expect(initialValues['bmi']).toBe(bmiField.value);
```

Comment:
It was better before. I might be wrong, but I think the point of @samuelmale 's question was not "you should pro grammatically get the value from the mock," which has resulted in a very unpleasant little block of code, but rather "you should create a situation in which it is obvious that the value cannot have been calculated." Which is to say, enter a new height or weight such that the calculated BMI should be different from the observed BMI field value (which comes from the encounter). It is not clear to me that this test would fail if the value were being calculated from the field values, which it should. As @samuelmale said, it is not clear that the BMI number in the encounter is different than what would be produced by the calculation.

You can leave in the code that extracts the value from the mock, or change it back to how it was, I don't care about that.